File: C++/35.Search Insert Position.cpp
#include <bits/stdc++.h>
using namespace std;
/*
35.

*/
class Solution
{
private:
public:
    Solution();
    ~Solution();
    int searchInsert(vector<int>& nums,int target)
    {
        //寻找数组是否含有target或者target位置
        for (int i=0; i<nums.size(); i++){
            if (nums[i]>=target)
                return i;
        }
        
        //若target大于数组所有值则返回数组长度
        return nums.size();
    }
};

Solution::Solution()
{
}

Solution::~Solution()
{
}

int main(int argv,char** argc){
    Solution* solution = new Solution();
    vector<int> nums({1,2,4,5});
    int target = 3;
    cout<<solution->searchInsert(nums,target)<<endl;
    delete solution;
    return 0;
}

File: Python/35.Search Insert Position.py
class Solution:
    def searchInsert(self, nums, target):
        if target not in nums:
            for i in range(len(nums)):
                if nums[i]>target:
                    return i
            return len(nums)
        else:
            return nums.index(target)

def main():
    '''
    test case:
    [1,2,4,5]
    7
    '''
    a=Solution()
    nums=[1,2,4,5]
    target=7
    print(Solution.searchInsert(a,nums,target))

if __name__ == '__main__':
    main()
